components/collator/src/comparison.rs

```diff
@@ -100,26 +105,37 @@ impl Collator {
             // the default, the code here that omits the explicit
             // variant if it matches the default should become
             // unnecessary.
-            let original_locale = locale;
-            let mut filtered_locale = alloc::borrow::Cow::Borrowed(locale);
-
-            let key = icu_locid::extensions_unicode_key!("co");
-            if let Some(variant) = original_locale.get_unicode_ext(&key) {
-                let zh = filtered_locale.language() == icu_locid::subtags_language!("zh");
-                let sv = filtered_locale.language() == icu_locid::subtags_language!("sv");
+            const KEY: Key = key!("co");
+            if let Some(variant) = locale.get_unicode_ext(&KEY) {
+                let zh = locale.language() == language!("zh");
+                let sv = locale.language() == language!("sv");
                 // Omit the explicit collation variant if it is the default.
                 // "standard" is the default for all languages except zh and sv.
-                if !((!zh && !sv && variant == value!("standard"))
+                if (!zh && !sv && variant == value!("standard"))
                     || (zh && variant == value!("pinyin"))
-                    || (sv && variant == value!("reformed")))
+                    || (sv && variant == value!("reformed"))
                 {
-                    filtered_locale.to_mut().set_unicode_ext(key, variant);
+                    // Remove -u-co- and all other extensions by cloning just the langid
+                    holder = DataLocale::from(locale.get_langid());
+                    &holder
+                } else {
+                    // Remove all other extensions by cloning the langid and adding -u-co-
+                    let mut locale = Locale::from(locale.get_langid());
```

Comment:
Nit: Can you do this without pulling in `Locale` please? Add a new function to `DataLocale` if necessary

Reply:
Why is this a problem? This is not "pulling in" any code, the from just sets a field.